src/control/server/instance.go

```diff
@@ -0,0 +1,324 @@
+	}
+}
+
+// MountScmDevice mounts the configured SCM device (DCPM or ramdisk emulation)
+// at the mountpoint specified in the configuration. If the device is already
+// mounted, the function returns nil, indicating success.
+func (srv *IOServerInstance) MountScmDevice() error {
+	scmCfg := srv.runner.Config.Storage.SCM
+
+	isMount, err := srv.ext.isMountPoint(scmCfg.MountPoint)
+	if err != nil && !os.IsNotExist(errors.Cause(err)) {
+		return errors.WithMessage(err, "failed to check SCM mount")
+	}
+	if isMount {
+		srv.log.Debugf("%s already mounted", scmCfg.MountPoint)
+		return nil
+	}
+
+	srv.log.Debugf("attempting to mount existing SCM dir %s\n", scmCfg.MountPoint)
+	mntType, devPath, mntOpts, err := getMntParams(scmCfg)
+	if err != nil {
+		return errors.WithMessage(err, "getting scm mount params")
+	}
+
+	srv.log.Debugf("mounting scm %s at %s (%s)...", devPath, scmCfg.MountPoint, mntType)
+	if err := srv.ext.mount(devPath, scmCfg.MountPoint, mntType, uintptr(0), mntOpts); err != nil {
+		return errors.WithMessage(err, "mounting existing scm dir")
+	}
+
+	return nil
+}
+
+// Start checks to make sure that the instance has a valid superblock before
+// performing any required NVMe preparation steps and launching a managed
+// daos_io_server instance.
+func (srv *IOServerInstance) Start(ctx context.Context, errChan chan<- error) error {
+	if srv.superblock == nil {
+		if err := srv.ReadSuperblock(); err != nil {
+			return errors.Wrap(err, "start failed; no superblock")
+		}
+	}
+	if err := srv.bdevProvider.PrepareDevices(); err != nil {
+		return errors.Wrap(err, "start failed; unable to prepare NVMe device(s)")
+	}
+	if err := srv.bdevProvider.GenConfigFile(); err != nil {
+		return errors.Wrap(err, "start failed; unable to generate NVMe configuration for SPDK")
+	}
+
+	return srv.runner.Start(ctx, errChan)
+}
+
+// NotifyReady receives a ready message from the running IOServer
+// instance.
+func (srv *IOServerInstance) NotifyReady(msg *srvpb.NotifyReadyReq) {
+	srv.log.Debugf("I/O server instance %d ready: %v", srv.Index, msg)
+
+	go func() {
+		srv.instanceReady <- msg
+	}()
+}
+
+// AwaitReady returns a channel which receives a ready message
+// when the started IOServer instance indicates that it is
+// ready to receive dRPC messages.
+func (srv *IOServerInstance) AwaitReady() chan *srvpb.NotifyReadyReq {
+	return srv.instanceReady
+}
+
+// NotifyStorageReady releases any blocks on AwaitStorageReady().
+func (srv *IOServerInstance) NotifyStorageReady() {
+	srv.log.Debugf("I/O server instance %d notifying storage ready", srv.Index)
+	go func() {
+		close(srv.storageReady)
+	}()
+}
+
+// AwaitStorageReady blocks until the IOServer's storage is ready.
+func (srv *IOServerInstance) AwaitStorageReady(ctx context.Context) {
+	select {
+	case <-ctx.Done():
+		srv.log.Debugf("I/O server instance %d storage not ready: %s", srv.Index, ctx.Err())
+	case <-srv.storageReady:
+		srv.log.Debugf("I/O server instance %d ready", srv.Index)
```

Comment:
this should read "storage ready" somewhere in the message

Reply:
done.

Reply:
Also, should these be at Info level instead of Debug? We need to be careful about not being too chatty, but these messages are only emitted at server startup so maybe it's OK to make them Info.

Reply:
I'm just going to make them Info.

Reply:
yes I think that's fine